Homework4/DiagonalMatrix/Models/DiagonalMatrix.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+
+namespace DiagonalMatrix
+{
+    public class DiagonalMatrix<T>
+    {
+        public delegate void ElementHandler(int indexI, int indexJ, T oldValue, T newValue);
```

Comment:
Requirements #5 

**>  Attention: use
> standard practices for working with events!**

The should be a class inherited from event args. 
https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern or check appropriate lection records